packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -126,46 +131,68 @@ class NetworkController {
         invalidRequests.add(requestData);
       }
     }
-    return HttpRequests(
+
+    currentValues.removeWhere((value) => value is WebSocket);
```

Comment:
nit: rather than removing all WebSocket, maybe just don't add them in the first place. it looks like we are adding them a couple lines earlier.

Reply:
At this point, `currentValues` contains all the current requests we have in the profiler, which will contain ws requests if they exist. When we refresh the network data after polling, the new data may contain web sockets with the same ids as ones we already have. For this reason, we need to remove the current web sockets and replace with the updated data in the profile. We could be smarter about this and keep unchanged sockets in the list if we need to.

Reply:
sg. can you add a comment to that effect?

Reply:
done